app/src/main/java/com/swent/assos/model/service/impl/DbServiceImpl.kt

```diff
@@ -54,6 +54,40 @@ constructor(
             } ?: emptyList())
   }
 
+  override suspend fun getUserByEmail(
+      email: String,
+      onSuccess: () -> Unit,
+      onFailure: () -> Unit
+  ): User {
+    val query = firestore.collection("users").whereEqualTo("email", email)
+    val snapshot = query.get().await()
+    if (snapshot.isEmpty) {
+      onFailure()
+      return User()
+    }
+    onSuccess()
+    return User(
+        id = snapshot.documents[0].id,
+        firstName = snapshot.documents[0].getString("firstname") ?: "",
+        lastName = snapshot.documents[0].getString("name") ?: "",
+        email = snapshot.documents[0].getString("email") ?: "",
+        following =
+            (snapshot.documents[0].get("following") as? MutableList<String>) ?: mutableListOf(),
+        associations =
+            snapshot.documents[0].get("associations")?.let { associations ->
+              (associations as? List<Map<String, Any>>)?.mapNotNull {
+                val assoId = it["assoId"] as? String
+                val position = it["position"] as? String
+                val rank = (it["rank"] as? Long)?.toInt()
+                if (assoId != null && position != null && rank != null) {
+                  Triple(assoId, position, rank)
+                } else {
+                  null
+                }
+              } ?: emptyList()
+            } ?: emptyList())
```

Comment:
Think about refactoring this code into a `serialise` function. It will be used somewhere else in the future so we will avoid duplicated code as much as possible